### PR TITLE
Fix problem with right-clicking on a right-to-left selection

### LINKED ISSUE #80119: SelectableText focus issue
## Steps to Reproduce

1. Run `flutter create bug`.
2. Update the files as follows:
use SelectableText instead of the Text widget with 'You have pushed the button this many times:' (main.dart:96)
3. `flutter run` targetting web (I used Chrome)
4. Select more than a word in the SelectableText from **right** to **left**
5. Click with the secondary mouse button on the selected part

**Expected results:**
The selection will remain the same

**Actual results:**
The selection was cropped to fit only one word. Please, see the video:

https://user-images.githubusercontent.com/37451880/114197631-367a7200-995b-11eb-80ba-727766cf86f3.mp4

**My thoughts**
I found `_lastSecondaryTapWasOnSelection` getter in flutter/lib/src/widgets/text_selection.dart which is used in `onSecondaryTap`  to handle mouse click. It returns whether textPosition.offset is inside current selection (as far as i understood) 
```dart
return renderEditable.selection!.base.offset <= textPosition.offset
        && renderEditable.selection!.extent.offset >= textPosition.offset;
```
It assumes that renderEditable.selection!.base.offset is less than renderEditable.selection!.extent.offset which is not true, according to base and extent documentation. So, maybe, if we change this part of code to something like
```dart
final minOffset = math.min(renderEditable.selection!.base.offset,
    renderEditable.selection!.extent.offset);
final maxOffset = math.max(renderEditable.selection!.base.offset,
    renderEditable.selection!.extent.offset);

return minOffset <= textPosition.offset && textPosition.offset <= maxOffset;
```
it will handle correctly this case.

<details>
  <summary>Logs</summary>

```
>flutter run --verbose  
[  +99 ms] executing: [C:\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[ +131 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +2 ms] b1395592de68cc8ac4522094ae59956dd21a91db
[   +1 ms] executing: [C:\flutter/] git tag --points-at b1395592de68cc8ac4522094ae59956dd21a91db
[ +320 ms] Exit code 0 from: git tag --points-at b1395592de68cc8ac4522094ae59956dd21a91db
[   +1 ms] 2.0.4
[  +62 ms] executing: [C:\flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +86 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [C:\flutter/] git ls-remote --get-url origin
[  +70 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[ +153 ms] executing: [C:\flutter/] git rev-parse --abbrev-ref HEAD
[  +65 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[ +114 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +79 ms] executing: C:\Users\Dmitry\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[ +111 ms] List of devices attached
[   +9 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +4 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +97 ms] Multiple devices found:
[  +38 ms] Chrome (web) • chrome • web-javascript • Google Chrome 89.0.4389.114
[   +3 ms] Edge (web)   • edge   • web-javascript • Microsoft Edge 89.0.774.57
[   +2 ms] [1]: Chrome (chrome)
[        ] [2]: Edge (edge)
[   +3 ms] Please choose one (To quit, press "q/Q")
[        ] : 
[+4011 ms] 1
[  +18 ms] Skipping pub get: version match.
[ +155 ms] Generating C:\Users\Dmitry\Documents\BitBucket\flutter_sandbox\bug\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[ +117 ms] Launching lib\main.dart on Chrome in debug mode...
[ +166 ms] Updating assets
[ +121 ms] Waiting for connection from debug service on Chrome...
[ +113 ms] <- reset
[  +11 ms] C:\flutter\bin\cache\dart-sdk\bin\dart.exe --disable-dart-dev C:\flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot      
--sdk-root C:\flutter\bin\cache\flutter_web_sdk/ --incremental --target=dartdevc --debugger-module-names --experimental-emit-debug-metadata
-DFLUTTER_WEB_AUTO_DETECT=true --output-dill C:\Users\Dmitry\AppData\Local\Temp\flutter_tools.5225a04d\flutter_tool.1e76d96d\app.dill --libraries-spec        
file:///C:/flutter/bin/cache/flutter_web_sdk/libraries.json --packages C:\Users\Dmitry\Documents\BitBucket\flutter_sandbox\bug\.dart_tool\package_config.json 
-Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-root
C:\Users\Dmitry\AppData\Local\Temp\flutter_tools.5225a04d\flutter_tools.6c075e1b --filesystem-scheme org-dartlang-app --initialize-from-dill
build\c7922d95bf4a2462b75c84a97c312edb.cache.dill.track.dill --platform file:///C:/flutter/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk.dill
--no-sound-null-safety
[  +19 ms] <- compile org-dartlang-app:/web_entrypoint.dart
[+19418 ms] Waiting for connection from debug service on Chrome... (completed in 19,5s)
[   +1 ms] Synced 29.1MB.       
[   +1 ms] <- accept
[   +1 ms] Caching compiled dill
[ +407 ms] Using Окно или вкладка откроются в текущем сеансе браузера.

[+1470 ms] [CHROME]:
[   +2 ms] [CHROME]:DevTools listening on ws://127.0.0.1:50348/devtools/browser/202b0c57-00f1-47cc-a9ac-ffbddb736dfe
[+1621 ms] DwdsInjector: Received request for entrypoint at http://localhost:50307/main_module.bootstrap.js
[  +11 ms] MetadataProvider: Loading debug metadata...
[  +23 ms] MetadataProvider: Loaded debug metadata
[   +9 ms] DwdsInjector: Injected debugging metadata for entrypoint at http://localhost:50307/main_module.bootstrap.js
[+8474 ms] DevHandler: Debug service listening on ws://127.0.0.1:50390/9vntKpLHa0c=/ws

[  +17 ms] Debug service listening on ws://127.0.0.1:50390/9vntKpLHa0c=/ws
[        ] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[   +2 ms]   To hot restart changes while running, press "r" or "R".
[        ] For a more detailed help message, press "h". To quit, press "q".
```

```
>flutter analyze
Analyzing bug...                                                        
No issues found! (ran in 5.0s)
```

```
C:\Users\Dmitry\Documents\BitBucket\flutter_sandbox\bug>flutter doctor -v          
[√] Flutter (Channel stable, 2.0.4, on Microsoft Windows [Version 10.0.19042.867], locale ru-RU)
    • Flutter version 2.0.4 at C:\flutter
    • Framework revision b1395592de (8 days ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\Dmitry\AppData\Local\Android\sdk
    • Platform android-30, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 41.1.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] IntelliJ IDEA Community Edition (version 2018.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2018.2
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code, 64-bit edition (version 1.54.3)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.0.2

[√] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 89.0.4389.114
    • Edge (web)   • edge   • web-javascript • Microsoft Edge 89.0.774.57

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by ludwiktrammer**
Fixes #80119, #79592. Right clicking on a selection that was created from right-to-left will now work the same as left clicking on a selection that was created from left-to-right.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/widgets/text_selection_test.dart [hunk 1]
```diff
@@ -487,6 +487,48 @@ void main() {
     expect(renderEditable.selectPositionAtCalled, isTrue);
   });
 
+    testWidgets('TextSelectionGestureDetectorBuilder right click', (WidgetTester tester) async {
```
Comment (LongCatIsLooong) [inline packages/flutter/test/widgets/text_selection_test.dart:490]: nit: indentation

Comment (LongCatIsLooong) [reply packages/flutter/test/widgets/text_selection_test.dart:490]: Also could you add a comment to this test that says it's a regression test for https://github.com/flutter/flutter/issues/80119, for posterity?

**Review by LongCatIsLooong — APPROVED** [review]
LGTM thanks for the fix!

**Review by chunhtai — APPROVED** [review]
LGTM

File: packages/flutter/test/widgets/text_selection_test.dart (hunk 1 above)
Comment (ludwiktrammer) [reply packages/flutter/test/widgets/text_selection_test.dart]: Thank you for the review! Both issues are now fixed.

**Comment by ithinkso117** [comment]
What the status of this PR?  We need this PR.... Please...

**Comment by LongCatIsLooong** [comment]
I'm not sure if the test failures (especially the customer testing failures) are flakes. @ludwiktrammer could you sync the PR to `HEAD` and push again, see if the issues persist?

**Comment by ludwiktrammer** [comment]
@LongCatIsLooong Done. This time the checks have passed successfully :)

**Pull request merged**